src/ILCompiler.WebAssembly/src/Compiler/DependencyAnalysis/WebAssemblyVTableSlotNode.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+using Internal.Runtime;
+using Internal.Text;
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    public class WebAssemblyVTableSlotNode : ObjectNode, ISymbolDefinitionNode
+    {
+        private readonly MethodDesc _targetMethod;
+
+        public WebAssemblyVTableSlotNode(MethodDesc targetMethod)
+        {
+            Debug.Assert(targetMethod.IsVirtual);
+            Debug.Assert(!targetMethod.IsSharedByGenericInstantiations);
+            _targetMethod = targetMethod;
+        }
+
+        public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
+        {
+            sb.Append(GetMangledName(nameMangler, _targetMethod));
+        }
+        public int Offset => 0;
+
+        public override bool IsShareable => false;
+
+        public static string GetMangledName(NameMangler nameMangler, MethodDesc method)
+        {
+            return "__getslot__" + nameMangler.GetMangledMethodName(method);
+        }
+
+        protected override string GetName(NodeFactory factory) => this.GetMangledName(factory.NameMangler);
+
+        public override ObjectNodeSection Section => ObjectNodeSection.DataSection;
+
+        public override bool StaticDependenciesAreComputed => true;
+
+        protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFactory factory)
+        {
+            DependencyList result = new DependencyList();
+
+            if (!factory.VTable(_targetMethod.OwningType).HasFixedSlots)
+            {
+                result.Add(factory.VirtualMethodUse(_targetMethod), "VTable method use");
+            }
+
+            return result;
+        }
+
+        public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
+        {
+
+            ObjectDataBuilder objData = new ObjectDataBuilder(factory, relocsOnly);
+
+            objData.AddSymbol(this);
+
+            if (!relocsOnly)
+            {
+                var tableOffset = EETypeNode.GetVTableOffset(factory.Target.PointerSize) / factory.Target.PointerSize;
+                objData.EmitInt(tableOffset + VirtualMethodSlotHelper.GetVirtualMethodSlot(factory, _targetMethod));
```

Comment:
Is this multiplied by pointer size at runtime? It might be better to just save it premultiplied by pointer size so that this is a simple addition at runtime.

Reply:
storing it as a multiple of pointers allows the LLVM/WASM instructions to be shorter because we treat the EEType as a blob of function pointers and this lets us directly index without having to cast anything.